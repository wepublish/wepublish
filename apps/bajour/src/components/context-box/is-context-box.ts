import {ApiV1, isBreakBlock} from '@wepublish/website'
import {allPass} from 'ramda'

const hasStyle = (blockStyle: string) => (block: ApiV1.Block) => block.blockStyle === blockStyle

export const isContextBox = (block: ApiV1.Block): block is ApiV1.LinkPageBreakBlock =>
  allPass([hasStyle('ContextBox'), isBreakBlock])(block)
