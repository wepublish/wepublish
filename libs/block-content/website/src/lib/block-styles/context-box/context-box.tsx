import { css } from '@mui/material';
import styled from '@emotion/styled';
import { useReducer } from 'react';
import { MdInfoOutline } from 'react-icons/md';
import { RichTextBlock } from '../../richtext/richtext-block';
import { allPass } from 'ramda';
import { hasBlockStyle } from '../../has-blockstyle';
import { isBreakBlock } from '../../break/break-block';
import {
  BuilderBlockStyleProps,
  useWebsiteBuilder,
} from '@wepublish/website/builder';
import { BlockContent, BreakBlock } from '@wepublish/website/api';

export const ContextBoxWrapper = styled('aside')`
  display: grid;
  grid-template-columns: 36px 1fr;
  grid-auto-rows: max-content;
  gap: ${({ theme }) => theme.spacing(1)};
  align-items: stretch;
  padding: ${({ theme }) => theme.spacing(1)};
`;

export const ContextBoxAllAbout = styled('div')`
  align-self: center;
  font-weight: 600;
`;

export const ContextBoxTitle = styled('div')`
  font-weight: 500;
  font-style: italic;
`;

export const ContextBoxCollapse = styled('div')<{ expanded: boolean }>`
  height: ${({ theme }) => theme.spacing(15)};
  overflow: hidden;
  interpolate-size: allow-keywords;
  transition: height 0.3s ease-in-out;
  font-style: italic;

  ${({ theme, expanded }) =>
    expanded &&
    css`
      height: auto;
    `}
`;

export const ContextBoxContent = styled('div')`
  display: flex;
  flex-flow: column;
  align-items: start;
  gap: ${({ theme }) => theme.spacing(1)};
`;

export const ContextBoxLine = styled('div')`
  justify-self: center;
  width: ${({ theme }) => theme.spacing(1)};
  background-color: currentColor;
`;

export const ContextBoxIcon = styled(MdInfoOutline)``;

export const ContextBox = ({
  className,
  richText,
  text,
}: BuilderBlockStyleProps['ContextBox']) => {
  const [expanded, toggleExpanded] = useReducer(exp => !exp, false);
  const {
    elements: { Button },
  } = useWebsiteBuilder();

  return (
    <ContextBoxWrapper className={className}>
      <ContextBoxIcon size="36" />

      <ContextBoxTitle>{text || `Darum geht's`}</ContextBoxTitle>

      <ContextBoxLine />
      <ContextBoxContent>
        <ContextBoxCollapse expanded={expanded}>
          <RichTextBlock richText={richText} />
        </ContextBoxCollapse>

        <Button onClick={toggleExpanded}>
          {expanded ? 'Weniger lesen' : 'Alles lesen'}
        </Button>
      </ContextBoxContent>
    </ContextBoxWrapper>
  );
};

export const isContextBoxBlockStyle = (
  block: Pick<BlockContent, '__typename'>
): block is BreakBlock =>
  allPass([hasBlockStyle('ContextBox'), isBreakBlock])(block);
