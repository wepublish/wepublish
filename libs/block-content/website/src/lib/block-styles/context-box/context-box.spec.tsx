import {render} from '@testing-library/react'
import * as stories from './context-box.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('Context Box', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      render(<Component />)
    })
  })
})
