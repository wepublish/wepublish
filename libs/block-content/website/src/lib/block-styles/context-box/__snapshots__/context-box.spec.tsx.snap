// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context Box should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-columns: 36px 1fr;
  grid-auto-rows: max-content;
  gap: 8px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  padding: 8px;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-weight: 600;
}

.emotion-6 {
  justify-self: center;
  width: 8px;
  background-color: currentColor;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 8px;
}

.emotion-10 {
  font-weight: 500;
  font-style: italic;
}

.emotion-12 {
  height: 120px;
  max-height: 120px;
  overflow: hidden;
  -webkit-transition: max-height 0.3s ease-in-out;
  transition: max-height 0.3s ease-in-out;
  font-style: italic;
}

.emotion-16 {
  margin: 0;
  line-height: 1.7;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  margin-bottom: 0.35em;
}

.emotion-16.MuiTypography-gutterBottom {
  margin-bottom: 24px;
}

.emotion-16:last-child {
  margin-bottom: 0;
}

.emotion-17 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  line-height: 1.4;
  margin-bottom: 0.35em;
}

.emotion-17:last-child {
  margin-bottom: 0;
}

.emotion-20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.4;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #e91e63;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  border-radius: 30px;
  text-transform: initial;
  box-shadow: none;
  font-weight: 700;
}

.emotion-20::-moz-focus-inner {
  border-style: none;
}

.emotion-20.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-20 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-20:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(163, 21, 69);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-20:hover {
    background-color: #e91e63;
  }
}

.emotion-20:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-20.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-20.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-20:hover {
  box-shadow: none;
}

.emotion-21 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<aside
    class="emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      height="36"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <div
      class="emotion-4 emotion-5"
    >
      Darum geht’s:
    </div>
    <div
      class="emotion-6 emotion-7"
    />
    <div
      class="emotion-8 emotion-9"
    >
      <div
        class="emotion-10 emotion-11"
      >
        Break block test
      </div>
      <div
        class="emotion-12 emotion-13"
      >
        <div
          class="emotion-14 emotion-15"
        >
          <div
            data-gramm="false"
            data-slate-editor="true"
            data-slate-node="value"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </p>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </h4>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-length="0"
                  data-slate-zero-width="n"
                >
                  ﻿
                  <br />
                </span>
              </span>
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <strong
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                       Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </span>
                  </strong>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit
                  </span>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <u
                  data-slate-leaf="true"
                >
                  <em
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                    </span>
                  </em>
                </u>
              </span>
            </p>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-20"
        tabindex="0"
        type="button"
      >
        Alles lesen
        <span
          class="MuiTouchRipple-root emotion-21"
        />
      </button>
    </div>
  </aside>
</DocumentFragment>
`;

exports[`Context Box should render WithClassName 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-columns: 36px 1fr;
  grid-auto-rows: max-content;
  gap: 8px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  padding: 8px;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-weight: 600;
}

.emotion-6 {
  justify-self: center;
  width: 8px;
  background-color: currentColor;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 8px;
}

.emotion-10 {
  font-weight: 500;
  font-style: italic;
}

.emotion-12 {
  height: 120px;
  max-height: 120px;
  overflow: hidden;
  -webkit-transition: max-height 0.3s ease-in-out;
  transition: max-height 0.3s ease-in-out;
  font-style: italic;
}

.emotion-16 {
  margin: 0;
  line-height: 1.7;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  margin-bottom: 0.35em;
}

.emotion-16.MuiTypography-gutterBottom {
  margin-bottom: 24px;
}

.emotion-16:last-child {
  margin-bottom: 0;
}

.emotion-17 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  line-height: 1.4;
  margin-bottom: 0.35em;
}

.emotion-17:last-child {
  margin-bottom: 0;
}

.emotion-20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.4;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #e91e63;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  border-radius: 30px;
  text-transform: initial;
  box-shadow: none;
  font-weight: 700;
}

.emotion-20::-moz-focus-inner {
  border-style: none;
}

.emotion-20.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-20 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-20:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(163, 21, 69);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-20:hover {
    background-color: #e91e63;
  }
}

.emotion-20:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-20.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-20.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-20:hover {
  box-shadow: none;
}

.emotion-21 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<aside
    class="extra-classname emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      height="36"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <div
      class="emotion-4 emotion-5"
    >
      Darum geht’s:
    </div>
    <div
      class="emotion-6 emotion-7"
    />
    <div
      class="emotion-8 emotion-9"
    >
      <div
        class="emotion-10 emotion-11"
      >
        Break block test
      </div>
      <div
        class="emotion-12 emotion-13"
      >
        <div
          class="emotion-14 emotion-15"
        >
          <div
            data-gramm="false"
            data-slate-editor="true"
            data-slate-node="value"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </p>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </h4>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-length="0"
                  data-slate-zero-width="n"
                >
                  ﻿
                  <br />
                </span>
              </span>
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <strong
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                       Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </span>
                  </strong>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit
                  </span>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <u
                  data-slate-leaf="true"
                >
                  <em
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                    </span>
                  </em>
                </u>
              </span>
            </p>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-20"
        tabindex="0"
        type="button"
      >
        Alles lesen
        <span
          class="MuiTouchRipple-root emotion-21"
        />
      </button>
    </div>
  </aside>
</DocumentFragment>
`;

exports[`Context Box should render WithEmotion 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-columns: 36px 1fr;
  grid-auto-rows: max-content;
  gap: 8px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  padding: 8px;
}

.emotion-4 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-weight: 600;
}

.emotion-6 {
  justify-self: center;
  width: 8px;
  background-color: currentColor;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  gap: 8px;
}

.emotion-10 {
  font-weight: 500;
  font-style: italic;
}

.emotion-12 {
  height: 120px;
  max-height: 120px;
  overflow: hidden;
  -webkit-transition: max-height 0.3s ease-in-out;
  transition: max-height 0.3s ease-in-out;
  font-style: italic;
}

.emotion-16 {
  margin: 0;
  line-height: 1.7;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  margin-bottom: 0.35em;
}

.emotion-16.MuiTypography-gutterBottom {
  margin-bottom: 24px;
}

.emotion-16:last-child {
  margin-bottom: 0;
}

.emotion-17 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  line-height: 1.4;
  margin-bottom: 0.35em;
}

.emotion-17:last-child {
  margin-bottom: 0;
}

.emotion-20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Merriweather,Roboto,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.4;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #e91e63;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  border-radius: 30px;
  text-transform: initial;
  box-shadow: none;
  font-weight: 700;
}

.emotion-20::-moz-focus-inner {
  border-style: none;
}

.emotion-20.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-20 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-20:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(163, 21, 69);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-20:hover {
    background-color: #e91e63;
  }
}

.emotion-20:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-20.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-20.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-20:hover {
  box-shadow: none;
}

.emotion-21 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<aside
    class="emotion-0 emotion-1"
  >
    <svg
      class="emotion-2 emotion-3"
      fill="currentColor"
      height="36"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <div
      class="emotion-4 emotion-5"
    >
      Darum geht’s:
    </div>
    <div
      class="emotion-6 emotion-7"
    />
    <div
      class="emotion-8 emotion-9"
    >
      <div
        class="emotion-10 emotion-11"
      >
        Break block test
      </div>
      <div
        class="emotion-12 emotion-13"
      >
        <div
          class="emotion-14 emotion-15"
        >
          <div
            data-gramm="false"
            data-slate-editor="true"
            data-slate-node="value"
            style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </p>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </span>
            </h4>
            <h4
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom emotion-17"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-length="0"
                  data-slate-zero-width="n"
                >
                  ﻿
                  <br />
                </span>
              </span>
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-16"
              data-slate-node="element"
            >
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <strong
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                       Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </span>
                  </strong>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit
                  </span>
                </em>
              </span>
              <span
                data-slate-node="text"
              >
                <u
                  data-slate-leaf="true"
                >
                  <em
                    data-slate-leaf="true"
                  >
                    <span
                      data-slate-string="true"
                    >
                      Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                    </span>
                  </em>
                </u>
              </span>
            </p>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary emotion-20"
        tabindex="0"
        type="button"
      >
        Alles lesen
        <span
          class="MuiTouchRipple-root emotion-21"
        />
      </button>
    </div>
  </aside>
</DocumentFragment>
`;
